### ISSUE TITLE
Change Detection is not working with asynchronous route changes

### ISSUE THREAD

**Issue by akopchinskiy** [angular/angular#24294]
## I'm submitting a...
Bug report

## Current behavior
Change Detection is not working with asynchronous route change.
After route is changed - change detection in any component linked to a route is not firing. Method `ngOnInit()` is not launched too, its lauched only after any interaction with a component (form input, etc.).

## Expected behavior
Change Detection is working with asynchronous route change.

## Minimal reproduction of the problem with instructions
Sample:
https://stackblitz.com/edit/angular-tqcthi

## Environment
Angular version: 6.0.3

## Browser:
- [*] Chrome (desktop) version LATEST
 
## For Tooling issues:
- Node version: XX  
- Platform:  Windows

## Others:
typescript: 2.8.1

## Comments
It's working if Observable from `this.auth.logIn()` convert to a promise, like this:
```typescript
this.auth.logIn() // returns Promise
     .then((result) => {
                this.router.navigateByUrl('myApp');
            }).catch((error) => {
                this.router.navigateByUrl('login');
            });
```
But I don't want to use promises, I want to use the Observable! Why It's not working? >:E There is no objective reasons for that.

**Comment by vicb** [comment]
Please attach a minimal repro

**Comment by akopchinskiy** [comment]
@vicb Done: https://stackblitz.com/edit/angular-tqcthi

**Comment by splincode** [comment]
@akopchinskiy 
![image](https://user-images.githubusercontent.com/12021443/40961913-a4b87ef2-68ac-11e8-9b4f-025e33e69002.png)

![image](https://user-images.githubusercontent.com/12021443/40961934-c28db294-68ac-11e8-96f0-b5f4a87ce103.png)

why do you use a local host? on the server this will not work
change the example

**Comment by akopchinskiy** [comment]
@splincode That's absolutely does not interfere the sample. Code runs as it should.

**Comment by Borzilov** [comment]
@akopchinskiy what exactly should happen according to your logic?

**Comment by akopchinskiy** [comment]
@Borzilov I just updated the example to make it more similar to real situation. It's working as expected, but the same code (copy-paste) does not work in my app. And I have no idea why.

**Comment by splincode** [comment]
@vicb I think you can close the task, a person just does not understand

**Comment by jasonaden** [comment]
This issue doesn't provide the steps to reproduce, or the exact expectation you're looking for with the example provided. It would be much easier to understand if you simplified the example somewhat (remove the extra calls inside your auth service) and explained what you're actually looking for.

Also, I'm closing and locking this issue based on the [Code of Conduct](https://github.com/angular/angular/blob/master/CODE_OF_CONDUCT.md) and the lack of respectful communication within this issue.

If the issue persists, please create a new issue with minimal reproduction, and explain exactly what to do to reproduce the issue you're having and the expected result.
